lib/rubocop/config.rb

```diff
@@ -196,16 +197,18 @@ def base_dir_for_path_parameters
 
     def target_ruby_version
       @target_ruby_version ||=
-        if File.file?('.ruby-version')
-          @target_ruby_version_source = :dot_ruby_version
-
-          /(ruby-)?(?<ruby_version>\d.\d)/ =~ File.read('.ruby-version')
-
-          ruby_version.to_f if ruby_version
-        else
+        if for_all_cops['TargetRubyVersion']
           @target_ruby_version_source = :rubocop_yml
 
           for_all_cops['TargetRubyVersion']
+        elsif File.file?('.ruby-version') &&
+              /\A(ruby-)?(?<ruby_version>\d.\d)/ =~ File.read('.ruby-version')
```

Comment:
It’s probably a while yet before this would be a problem, but you might consider changing `.` to `\.` and both `\d` to `\d+` to allow for a Ruby version such as 2.10.x.

See also https://www.ruby-lang.org/en/news/2016/04/01/ruby-2-1-10-released/ – in particular the quote “Please test your applications and/or libraries for compatibility with two-digit version numbers.”